### DISCUSSION TITLE
libspdm package crashing

### DISCUSSION THREAD

**Discussion by PoojaGaur1708** [DMTF/libspdm#818] [Q&A] [answered]
Hello , i am using libspdm package to send spdm packet .

I follow spdm_client_init function.
But when it calls , my code crash in libspdm_acquire_sender_buffer function.
As debug , value in this field of strure which is function pointer , having value 0x0. So pls help me urgently  / suggest me where and how to fill the value in this function pointer.
Seems function is missing for this .

libspdm_return_t libspdm_acquire_sender_buffer (
    libspdm_context_t *spdm_context, size_t *max_msg_size, void **msg_buf_ptr)
{
    libspdm_return_t status;

    LIBSPDM_ASSERT (spdm_context->sender_buffer == NULL);
    LIBSPDM_ASSERT (spdm_context->sender_buffer_size == 0);
     status = spdm_context->acquire_sender_buffer (spdm_context, max_msg_size, msg_buf_ptr);  /**### /CRASH AT THIS POINT**
    if (status != LIBSPDM_STATUS_SUCCESS) {
        return status;
    }
    spdm_context->sender_buffer = *msg_buf_ptr;
    spdm_context->sender_buffer_size = *max_msg_size;
    return LIBSPDM_STATUS_SUCCESS;
}

**Comment by jyao1** [comment, marked as answer]
Seems duplicated with https://github.com/DMTF/libspdm/issues/819
